frontend/src/Components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,215 @@
+import classNames from 'classnames';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Manager, Popper, Reference } from 'react-popper';
+import Portal from 'Components/Portal';
+import { kinds, tooltipPositions } from 'Helpers/Props';
+import dimensions from 'Styles/Variables/dimensions';
+import { isMobile as isMobileUtil } from 'Utilities/browser';
+import styles from './Tooltip.css';
+
+export interface TooltipProps {
+  className?: string;
+  bodyClassName?: string;
+  anchor: React.ReactNode;
+  tooltip: string | React.ReactNode;
+  kind?: Extract<(typeof kinds.all)[number], keyof typeof styles>;
```

Comment:
I didn't anticipate the type `(typeof kinds.all)[number]` would be used so much (same for sizes)

Exporting it from `Kinds` under a simpler name might be a good idea now

Reply:
Good call, did the same for the others. Ended up extending the types a bit to avoid issues with them being treated as strings in some places.